homework/zadanie4_five_philosophers.cpp

```diff
@@ -4,48 +4,28 @@
 #include <chrono>
 #include <mutex>
 #include <string>
-#include <mutex>
+#include <random>
 
 using namespace std::chrono_literals;
 
 
 class CFork{
-public:
-    mutable std::mutex mtx_fork_;
-};
-
- 
-class CTable{
-
-    int numbers_of_forks_;
 
 public:
-    std::vector<CFork*> forks_pool_;
-
-    CTable(int number_of_forks) : numbers_of_forks_(number_of_forks){
-        for (size_t i=0; i < numbers_of_forks_; i++){
-            forks_pool_.push_back(new CFork());
-        }
-    }
-
-    ~CTable(){
-        for (int i = 0; i < numbers_of_forks_; i++){
-            delete forks_pool_[i];
-        }
-    }
+    mutable std::mutex mtx_fork_;
 };
 
 class CPhilosoph{
 
     std::string name_;
     std::string output_;
-    CTable& table_;
     CFork* left_fork_;
     CFork* right_fork_;
     int time=0;
```

Comment:
nieużywane?

Reply:
bylby potrzebny, gdyby to bylo wesele i kilkanascie stolow, wtedy po kazdym tancu, po kazdym kielonku wewnetrzna nawigacja natrafialaby na bugi, i taki adres do stolu by sie przydal np koledze, zeby wiedzial gdzie ma go odstawic:) uwazam, ze w tym przypadku jest nadprogramowy,